app/models/concerns/omniauthable.rb

```diff
@@ -0,0 +1,27 @@
+module Omniauthable
+  extend ActiveSupport::Concern
+
+  included do
+    devise :omniauthable, omniauth_providers: [:google_oauth2]
+  end
+
+  class_methods do
+    def from_omniauth(auth)
+      where(provider: auth.provider, uid: auth.uid, organization_id: Current.organization.id).first_or_create do |user|
```

Comment:
Hmm. If I am staff authenticating with OAut, is this not going to set an adopter role on my account when I authenticate with OAuth?

We should only set the adopter role on the user IF the user is new. We should not touch roles of users who already exist, no?

Reply:
Ah nevermind...my misunderstanding of how `first_or_create` works. The block is skipped if the user exists. Nice.